### PR TITLE
feat: Represent dataset in a map

### LINKED ISSUE #540: Represent Datasets on a Map
## Description
Feature to configure how shared files will be rpresented on a map

## Additional context

Spec: [Represent Datasets on a Map](https://docs.google.com/document/d/1QzzKLJ-1qagxpFTKsqj-dv3w-kEMfeyeOD-8FMyRdZw/edit)
Figma: https://www.figma.com/file/JLdn5XFGnjVZhl3JpZu6aB/Display-dataset-R1.4?node-id=0%3A1

### PR DESCRIPTION
**Pull request by josebui**
## Description

- Represent dataset in a map feature

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added
- [ ] English strings reviewed and copyedited
- [ ] Strings are localized
- [ ] Verified on mobile
- [ ] Verified on desktop
- [ ] Verified accessibility ([Accessibility - Web applications](https://docs.google.com/document/d/1VwPDyLqw7r_iISMgDZr1FdAmAqqOIIVIxQqsYA-HaHE/edit?usp=sharing))

### Related Issues
Fixes #540

### CODE REVIEW

File: src/sharedData/visualization/components/VisualizationWrapper.js [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+  VisualizationContext,
+  VisualizationContextProvider,
+} from 'sharedData/visualization/visualizationContext';
+
+import Visualization from './Visualization';
+
+const formatDate = (language, dateString) =>
+  new Intl.DateTimeFormat(language, { dateStyle: 'medium' }).format(
+    Date.parse(dateString)
+  );
```
Comment (paulschreiber) [inline src/sharedData/visualization/components/VisualizationWrapper.js:38]: issue: this function is written out three times. Move it to a utils file.

File: src/sharedData/visualization/components/VisualizationConfigForm/index.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  datasetConfig: {
+    dataColumns: {
+      option: '',
+      selectedColumns: Array(3).fill(''),
+    },
+  },
+  visualizeConfig: {
+    shape: 'circle',
+    size: 15,
+    color: '#A96F14',
```
Comment (paulschreiber) [inline src/sharedData/visualization/components/VisualizationConfigForm/index.js:25]: issue: this color should come from theme.js

**Review by paulschreiber — CHANGES_REQUESTED** [review]
You're also missing useDocumentTitle() on the visualization pages.

File: package.json [hunk 3]
```diff
@@ -40,9 +43,11 @@
     "react-leaflet-markercluster": "^3.0.0-rc1",
     "react-redux": "^8.0.2",
     "react-router-dom": "^6.3.0",
+    "use-debounce": "^8.0.4",
     "uuid": "^9.0.0",
     "web-vitals": "^3.0.1",
     "world-countries": "^4.0.0",
+    "xlsx": "https://cdn.sheetjs.com/xlsx-0.18.10/xlsx-0.18.10.tgz",
```
[the remark this answers is not in the document]
Comment (josebui) [reply package.json:50]: @paulschreiber this is the reason:

https://docs.sheetjs.com/docs/getting-started/installation/nodejs
 
image

File: src/sharedData/visualization/visualizationUtils.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+import * as SheetsJs from 'xlsx';
```
[the remark this answers is not in the document]
Comment (josebui) [reply src/sharedData/visualization/visualizationUtils.js:1]: @paulschreiber I think that can be a potential improvement, but right now that is not the way the other files like this are named, so I don't think we should do this change as part of this PR

[the remark this answers is not in the document]
Comment (paulschreiber) [reply src/sharedData/visualization/visualizationUtils.js:1]: I hadn't noticed that. Having visualizationXYZ inside visualization is redundant. We should fix them all.

File: src/sharedData/visualization/components/VisualizationConfigForm/SelectDataFileStep.js [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+        >
+          <li>Requirment 1</li>
+          <li>Requirment 2</li>
+        </Stack>
+        <Typography sx={{ mb: 2 }}>
+          Recommentation
+          <ExternalLink
+            href={t('sharedData.form_step_data_file_step_description_help_url')}
+          >
+            help link
```
[the remark this answers is not in the document]
Comment (josebui) [reply src/sharedData/visualization/components/VisualizationConfigForm/SelectDataFileStep.js:79]: @paulschreiber Can you add more context to this? Most of the strings still need to be localized to spanish

[the remark this answers is not in the document]
Comment (paulschreiber) [reply src/sharedData/visualization/components/VisualizationConfigForm/SelectDataFileStep.js:75]: You have the bare strings i.e. `Recommentation` and no `t()` call.

[the remark this answers is not in the document]
Comment (josebui) [reply src/sharedData/visualization/components/VisualizationConfigForm/SelectDataFileStep.js:75]: @paulschreiber this is wrapped with by a `Trans` component, this content is localized

File: src/sharedData/visualization/visualizationUtils.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (josebui) [reply src/sharedData/visualization/visualizationUtils.js:1]: @paulschreiber We can talk about that for future improvements, it is not a trivial decision, and defenitely not in the scope of this feature, that will be a lot of files changed outside of this

File: src/sharedData/visualization/components/VisualizationConfigForm/SelectDataFileStep.js [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+          first
+        </Typography>
+        <Stack
+          component="ul"
+          spacing={1}
+          sx={{ mb: 2 }}
+          aria-labelledby="visualization-file-requirements"
+        >
+          <li>Requirment 1</li>
+          <li>Requirment 2</li>
```
Comment (paulschreiber) [inline src/sharedData/visualization/components/VisualizationConfigForm/SelectDataFileStep.js:68]:
```suggestion
          <li>Requirement 1</li>
          <li>Requirement 2</li>
```

File: src/sharedData/visualization/components/VisualizationConfigForm/SelectDataFileStep.js [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+        </Stack>
+        <Typography sx={{ mb: 2 }}>
+          Recommentation
```
Comment (paulschreiber) [inline src/sharedData/visualization/components/VisualizationConfigForm/SelectDataFileStep.js:71]:
```suggestion
          Recommendation
```

**Review by paulschreiber — CHANGES_REQUESTED** [review]
Fix spelling of strings in <Trans>

File: src/localization/locales/en-US.json [hunk 8]
```diff
@@ -337,14 +342,85 @@
...
+        "form_step_visualize_shape_triangle": "Triangle",
+        "form_visualization_preview_label": "Preview",
+        "deleted_visualization_config": "{{mapTitle}} has been deleted.",
+        "form_step_annotate_data_points_input_label": "Data point {{index}} label",
+        "visualization_file_load_error": "The file “{{name}}” could not be loaded. Please try again or select another file.",
+        "form_step_set_dataset.form.validation_array_min": "Please select at least one data column to display.",
+        "map_create_document_title": "Create Map: {{name}}",
+        "map_view_document_title": "View Map: {{name}}",
+        "fetchVisualizationConfig.not_found": "Map not found",
+        "addVisualizationConfig.unique": "[TODO] The title “{{body.variables.input.title}}“ is being used by other map, change it and try again."
```
Comment (josebui) [inline src/localization/locales/en-US.json:423]: @paulschreiber Can you review this string? This is the message the user will see when the title is already being used by other map

Comment (paulschreiber) [inline src/localization/locales/en-US.json:423]:
```suggestion
        "addVisualizationConfig.unique": "The name “{{body.variables.input.title}}“ is in use by another map. Please use a different name."
```

**Review by paulschreiber — APPROVED** [review]

**Pull request merged**